quickwit/quickwit-query/src/query_ast/tantivy_query_ast.rs

```diff
@@ -881,14 +937,18 @@ mod tests {
             let filter = proptest::collection::vec(element.clone(), 0..4);
             let should = proptest::collection::vec(element.clone(), 0..4);
             let must_not = proptest::collection::vec(element.clone(), 0..4);
-            (must, filter, should, must_not).prop_map(|(must, filter, should, must_not)| {
-                TantivyQueryAst::Bool(TantivyBoolQuery {
-                    must,
-                    filter,
-                    should,
-                    must_not,
-                })
-            })
+            let minimum_should_match = (0usize..2).prop_map(|n: usize| n.checked_sub(1));
```

Comment:
if we generate some minimum_should_match, `evaluate_test` must be updated to to consider it. Otherwise we don't verify it on the generated nor simplified request, so the test tests nothing

Reply:
updated the evaluate_test... and found a bug! thx